### DISCUSSION TITLE
How to determine a cause of freeze

### DISCUSSION THREAD

**Discussion by AuracleTech** [ppy/osu#17255] [Q&A] [answered]
Anyone has an idea on how I could improve osu lazer by finding what cause freezes?
My CPU never reach full throttle and I have no clue how to find what could be causing it

https://user-images.githubusercontent.com/40531871/158295434-07775acd-8934-4f39-b8e0-3fbbcc9b5acc.mp4

**Comment by frenzibyte** [comment]
Start by re-recording the video with the frame graph displayed (Ctrl+F11 twice). Impossible to tell without that.

**Comment by AuracleTech** [comment]
https://user-images.githubusercontent.com/40531871/158302303-4b27bc85-6a16-4635-9587-4b0f2ea4585f.mp4

I don't know what I can do with with this to target the issue

**Reply by peppy** [reply]
From that I can tell you it's something else on your PC causing GPU pipeline stalls, and potentially fully event thread stalls.

Open task manager and maybe google around for information on how to figure out what is bogging your system down.

https://osu.ppy.sh/help/wiki/Performance_Troubleshooting may also be of use.

**Reply by AuracleTech** [reply]
I will try to pinpoint the issue
Note that this only happens in lazer which is why it really confuses me.

**Comment by AuracleTech** [comment, marked as answer]
After 10 minutes 14 seconds of playing lazer, I started recording after it happened off-recording, so very unlikely it would be OBS.
Those stutters does not occur in osu stable, League of Legends, Overwatch, Apex legends and LoL TFT.

https://user-images.githubusercontent.com/40531871/158914106-4138155d-759d-4726-b00b-8a31f1d93b98.mp4

**Reply by frenzibyte** [reply]
That aside, why are you running at this high framerate? (looks like 8x refresh rate on a 280hz monitor, which cranks up the update thread to 2240hz) I'm wondering if you can still get this spike on a lower frame rate, out of curiosity.

**Reply by AuracleTech** [reply]
My monitor is 280hz I play on I think x8? The setting just before max framerate. I will re-try later to check under 2x. or under

**Reply by frenzibyte** [reply]
That said, and although probably not of any importance, but are the other games you played also running at this massive framerate? I definitely recommend lowering that down to at the very most ~960fps maximum.

**Reply by AuracleTech** [reply]
Yeah no my other games never exceed 1K, I'll try lowering the max, maybe this should be a default if it causes issues
I'll try and game at under 1K fps

**Reply by frenzibyte** [reply]
The maximum by default *is* "2x refresh rate" in osu!(lazer).

**Reply by AuracleTech** [reply]
I understand but if going over a certain 'healthy' limit for the gpu actually causes stutter for some gpus, maybe a hard lock on more than 1000fps is necessary, On osu stable for example I can't go more than 960x so that might be the reason, it's kinda unexpected.

**Reply by AuracleTech** [reply]
After hours I can confirm that going over a certain frame rate can causes stutters
I recommend a hard lock or a warning when pushing fps cap too high
